packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -0,0 +1,442 @@
+  );
+  const { hasFrontend, hasBackend, hasBot, hasProvision } = await queryProjectStatus(fx);
+
+  //localSettings.json
+  const localSettingsProvider = new LocalSettingsProvider(projectPath);
+  await localSettingsProvider.save(localSettingsProvider.init(hasFrontend, hasBackend, hasBot));
+  //projectSettings.json
+  const projectSettings = path.join(fxConfig, ProjectSettingsFileName);
+  await fs.copy(path.join(fx, "settings.json"), projectSettings);
+  await ensureProgrammingLanguage(projectSettings, path.join(fx, "env.default.json"));
+  //config.dev.json
+  await fs.writeFile(
+    path.join(fxConfig, "config.dev.json"),
+    JSON.stringify(getConfigDevJson(await getAppName(projectSettings)), null, 4)
+  );
+  // appPackage
+  await fs.copy(path.join(projectPath, AppPackageFolderName), templateAppPackage);
+  await fs.rename(
+    path.join(templateAppPackage, "manifest.source.json"),
+    path.join(templateAppPackage, "manifest.template.json")
+  );
+
+  await moveIconsToResourceFolder(templateAppPackage);
+  if (hasProvision) {
+    const devProfile = path.join(fxPublishProfile, "profile.dev.json");
+    const devUserData = path.join(fxPublishProfile, "dev.userdata");
+    await fs.copy(path.join(fx, "new.env.default.json"), devProfile);
+    await fs.copy(path.join(fx, "default.userdata"), devUserData);
+    await removeExpiredFields(devProfile, devUserData);
+    await ensureActiveEnv(projectSettings);
+  }
+}
+async function moveIconsToResourceFolder(templateAppPackage: string): Promise<void> {
+  // see AppStudioPluginImpl.buildTeamsAppPackage()
+  const manifest: TeamsAppManifest = await readJson(
+    path.join(templateAppPackage, "manifest.template.json")
+  );
+  const hasColorIcon = manifest.icons.color && !manifest.icons.color.startsWith("https://");
+  const hasOutlineIcon = manifest.icons.outline && !manifest.icons.outline.startsWith("https://");
+  if (!hasColorIcon || !hasOutlineIcon) {
+    return;
+  }
+  // move to resources
+  const resource = path.join(templateAppPackage, "resources");
+  await fs.ensureDir(resource);
+  await fs.move(
+    path.join(templateAppPackage, manifest.icons.color),
+    path.join(resource, manifest.icons.color)
+  );
+
+  await fs.move(
+    path.join(templateAppPackage, manifest.icons.outline),
+    path.join(resource, manifest.icons.outline)
+  );
+  // update icons
+  manifest.icons.color = `resources/${manifest.icons.color}`;
+  manifest.icons.outline = `resources/${manifest.icons.outline}`;
+  await fs.writeFile(
+    path.join(templateAppPackage, "manifest.template.json"),
+    JSON.stringify(manifest, null, 4)
+  );
+}
+
+async function removeExpiredFields(devProfile: string, devUserData: string): Promise<void> {
+  // remove fx-resource-local-debug.trustDevCert, solution.programmingLanguage, fx-resource-function.defaultFunctionName
+  const profileData = await readJson(devProfile);
+  if (profileData[PluginNames.LDEBUG]) {
+    delete profileData[PluginNames.LDEBUG];
+  }
```

Comment:
Besides the configs under `fx-resource-local-debug`, we also need to clean up the local debug related config for other resource plugins (e.g. AAD, bot, simpleAuth, etc.) in `env.default.json`

Just for your reference:
```json
"fx-resource-aad-app-for-teams": {
    "local_clientId": "72613b3f-2381-45af-b7f8-7db9184c9fe2",
    "local_clientSecret": "{{fx-resource-aad-app-for-teams.local_clientSecret}}",
    "local_objectId": "41bacd6d-4b6f-412c-8a2c-be72fc4188ae",
    "local_oauth2PermissionScopeId": "52226346-7f9c-4524-8103-e60065977ea5",
    "local_tenantId": "9cebf8c7-7a35-483e-ba9f-1e1cb83461aa",
    "local_applicationIdUris": "api://localhost/botid-3e6b68ba-e2f5-4dab-8452-a2fa2b5de013"
},

"fx-resource-simple-auth": {
    "filePath": "{{fx-resource-simple-auth.filePath}}",
    "environmentVariableParams": "{{fx-resource-simple-auth.environmentVariableParams}}"
},

"fx-resource-bot": {
    "localBotId": "3e6b68ba-e2f5-4dab-8452-a2fa2b5de013",
    "localBotPassword": "{{fx-resource-bot.localBotPassword}}",
    "localObjectId": "afdd8b19-af00-4e85-aa21-f0c76d9bf9d0",
    "local_redirectUri": "https://c52b5493cab3.ngrok.io/public"
},
```

Reply:
Thanks, will remove.